bindings/py/cpp_src/bindings/algorithms/py_HTM.cpp

```diff
@@ -127,7 +127,7 @@ namespace nupic_ext
             return py::array_t<nupic::UInt32>(activeCells.size(), activeCells.data());
         });
 
-        py_HTM.def("getPredictiveCells", [](const HTM_t& self)
+        py_HTM.def("getPredictiveCells", [](HTM_t& self)
```

Comment:
I'm not a fan of getters not being const (like the stuff with SDRs), why is this needed?

Reply:
> When the active cells changes, the flag is cleared. When the active/matching segments are needed they are computed, and the flag is set. 

so could the getters stay as is (const) and add a new version updatePredictiveCells which would call activateDendrites? 

That way get* is fast, and you add your functionality (even w/o API change)

Reply:
If the getters are constant and they encounter `segmentsValid_ == false` then they will by necessity raise an exception.  The error message will explain that the user needs to call the method `updatePredictiveCells` before using the getters.